python_modules/dagster/dagster/_core/definitions/input.py

```diff
@@ -387,7 +405,29 @@ def maps_to_fan_in(self) -> bool:
 
     def describe(self) -> str:
         idx = self.maps_to.fan_in_index if isinstance(self.maps_to, FanInInputPointer) else ""
-        return f"{self.definition.name} -> {self.maps_to.solid_name}:{self.maps_to.input_name}{idx}"
+        return f"{self.in_name} -> {self.maps_to.solid_name}:{self.maps_to.input_name}{idx}"
+
+    def get_definition(self) -> "InputDefinition":
+        return InputDefinition(
+            name=self.in_name, description=self.description, dagster_type=self.dagster_type
+        )
+
+    @staticmethod
+    def create(
```

Comment:
Why have a `create` method instead of having the constructor accept these arguments?

Reply:
after our conversation, moved to just creating via constructor.